### PR TITLE
Minor discrepancy fixes

### PR DESCRIPTION
**Pull request by krmahadevan**
Making the codebase a bit more better place :)

### CODE REVIEW

File: testng-core/src/main/java/org/testng/xml/LaunchSuite.java [hunk 1]
```diff
@@ -272,7 +251,7 @@ static class ClassesAndMethodsSuite extends CustomizedSuite {
         final Map<String, String> parameters,
         final String annotationType,
```
Comment (juherr) [inline testng-core/src/main/java/org/testng/xml/LaunchSuite.java:252]: Could be removed too

File: testng-core/src/main/java/org/testng/xml/LaunchSuite.java [hunk 2]
```diff
@@ -345,7 +324,7 @@ static class ClassListSuite extends CustomizedSuite {
...
```
Comment (juherr) [inline testng-core/src/main/java/org/testng/xml/LaunchSuite.java:325]: Could be removed

File: testng-core/src/test/java/test/BaseTest.java [hunk 3]
```diff
@@ -309,27 +303,15 @@ public void setSuite(XmlSuite suite) {
...
-
-  protected static void verifyInstanceNames(
-      Map<String, List<ITestResult>> actual, String[] expected) {
-    List<String> actualNames = Lists.newArrayList();
-    for (Map.Entry<String, List<ITestResult>> es : actual.entrySet()) {
-      for (ITestResult tr : es.getValue()) {
-        Object instance = tr.getInstance();
-        actualNames.add(es.getKey() + "#" + (instance != null ? instance.toString() : ""));
-      }
+      assert expected == 0 : "Expected " + expected + " " + message + " but found " + 0;
```
Comment (juherr) [inline testng-core/src/test/java/test/BaseTest.java:313]: Legacy code: replace assert keyword by an assert method.

File: testng-core/src/test/java/test/CommandLineTest.java [hunk 4]
```diff
@@ -10,7 +9,7 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import org.jetbrains.annotations.Nullable;
+import javax.annotation.Nullable;
```
Comment (juherr) [inline testng-core/src/test/java/test/CommandLineTest.java:12]: javax.annotation should not be used.
I don't know what is the best alternative

**Review by juherr — APPROVED** [review]

File: testng-core/src/main/java/org/testng/xml/LaunchSuite.java (hunk 1 above)
Comment (krmahadevan) [reply testng-core/src/main/java/org/testng/xml/LaunchSuite.java:252]: I want to revisit a bunch of classes in this `org.testng.xml` package and perhaps deprecate them or remove them directly based on my analysis. I will do this in a separate PR.

File: testng-core/src/main/java/org/testng/xml/LaunchSuite.java (hunk 2 above)
Comment (krmahadevan) [reply testng-core/src/main/java/org/testng/xml/LaunchSuite.java:325]: I want to revisit a bunch of classes in this `org.testng.xml` package and perhaps deprecate them or remove them directly based on my analysis. I will do this in a separate PR.

**Pull request merged**